modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -39,13 +39,110 @@
 import java.io.IOException;
 import java.text.SimpleDateFormat;
 import java.util.*;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static org.openapitools.codegen.utils.StringUtils.camelize;
 import static org.openapitools.codegen.utils.StringUtils.underscore;
 
 public abstract class AbstractTypeScriptClientCodegen extends DefaultCodegen implements CodegenConfig {
+
+    @SuppressWarnings("StringBufferField")
+    public static class ParameterExpander {
```

Comment:
Btw: is there a policy that Code-Generators may not depend on other classes?

I'd rather have this a standalone class in package (e.g.) org.openapitools.codegen or .languages but refrained since this would be the only class that is not a Code-Generator of one of the abstract base classes

Reply:
I am not aware of a policy that would disallow depending on other classes. 

we can keep it in the abstract class, but if you want, we could create a sub package typescriptutils or something like that, and move it to a standalone class there. Up to you (you could do that in a separate PR)